### ISSUE TITLE
Meeseeks spec issues, a number of returns are not part of the public specs

### ISSUE THREAD

**Issue by OvermindDL1** [mischov/meeseeks#34]
Running dialyzer over a project that uses meeseeks shows some issues, listed below:

### First

This code:
```elixir
title = Meeseeks.one(doc, css("meta[property='og:title']"))
description = title && Meeseeks.one(doc, css("meta[property='og:description']"))
```
Generates this error on the `description = ...` line:
```elixir
lib/site_parser.ex:116: Guard test _@1::{'error',binary()} | #{'__struct__':='Elixir.Meeseeks.Result', 'document':=#{'__struct__':='Elixir.Meeseeks.Document', 'id_counter':='nil' | integer(), 'nodes':=#{integer()=>#{'__struct__':=atom(), atom()=>_}}, 'roots':=[integer()]}, 'id':=integer()} =:= 'false' can never succeed
```
In essence it's saying that it does not return a `nil` yet it does indeed in this case return a `nil`, thus the spectype for `Meeseeks.one` is incorrect.

### Second

This code:
```elixir
with(qt when qt != nil <- Meeseeks.one(doc, css(".qt"))) do
  qt
  |> Meeseeks.tree()
  |> case do
    {"p", [{"class", "qt"}], children} ->
      ...
    _ ->
      nil
  end
end
```
Generates this error on the `{"p", [{"class", "qt"}], children} ->` line:
```elixir
lib/site_parser.ex:100: The pattern {<<112:8/integer-unit:1>>, [{<<99:8/integer-unit:1,108:8/integer-unit:1,97:8/integer-unit:1,115:8/integer-unit:1,115:8/integer-unit:1>>, <<113:8/integer-unit:1,116:8/integer-unit:1>>}], Vchildren@1} can never match the type 'nil' | [binary() | {'comment',binary()} | {'pi',binary()} | {'pi' | binary(),binary() | [{binary(),binary()}],binary() | [any()]} | {'doctype',binary(),binary(),binary()}]
```
In essence it is saying that being tested is a `{binary(), [{binary(), binary()}], Vchildren@1}` is not a valid return type of `Meeseeks.tree()`, and yet it is because this successfully matches without issue.

**Comment by mischov** [comment]
Thanks. I'll get these fixed.

**Comment by mischov** [comment]
@OvermindDL1 
I'll fix the missing possibility of `nil` on `Meeseeks.one`, and I might have a fix for the second problem but I'm having a lot of problems knowing because I don't fully understand the problem, and when running Dialyzer with `-Wspecdiffs` I'm getting what appear to be just wrong success-types.

Could you indicate with what options you're running Dialyzer with and any other information I could use to better reproduce the issue?

**Comment by mischov** [comment]
@OvermindDL1 
Give master a shot and see if that solves the problems.

**Comment by OvermindDL1** [comment]
Cool, grabbed, rebuilding PLT's, waiting.... waiting... waiting... ... waiting..., testing, and success, no errors reported!  :-)

**Comment by mischov** [comment]
Fixed in v0.8.0.
